lib/group_domain_models/lib/src/models/school.dart

```diff
@@ -38,19 +39,19 @@ class School {
 }
 
 class SchoolData {
-  final String id, name, title, description, abbreviation;
-  final String publicKey, joinLink;
+  final String? id, name, title, description, abbreviation;
```

Comment:
Can't you make id non-nullable? (Are we even using SchoolData? I don't think so, right?)